src/Airship/Internal/Decision.hs

```diff
@@ -559,12 +559,17 @@ l13 r@Resource{..} = do
         Nothing ->
             m16 r
 
-l07 r = do
+l07 r@Resource{..} = do
     trace "l07"
     req <- lift request
-    if requestMethod req == HTTP.methodPost
-        then m07 r
-        else lift $ halt HTTP.status404
+    if requestMethod req == HTTP.methodPost then m07 r
+    else do
+        m <- _contentType <$> get
+        when (isJust m) $ do
+            let (Just (ct, body)) = m
+            lift body >>= lift . putResponseBody
+            lift $ addResponseHeader ("Content-Type", renderHeader ct)
```

Comment:
Yeah I noticed this in my alternative resource [PR](https://github.com/helium/airship/pull/82/files#diff-b5f02aff231ab03013a5644893829841R104).

So I just wanted to confirm that we want to use the action/body of `contentTypeProvided` here? Obviously that would make the basic example work as it's implemented.

If it is intentional, should we do the exact same thing as `o18` and default to the (unsafe) `head` of the list if no `Accept` header is provided? In that case I would definitely advocate extracting a common function to share between the two. In additional would it make sense to update haskell doc of that function in `Resource.hs`?

In not, and we just want to use the current `responseBody`, I suspect having a common function for setting the `Content-Type` might come in handy for the other non-200 responses.

I'm not 100% sure what the [erlang version](https://github.com/webmachine/webmachine/blob/master/src/webmachine_error_handler.erl#L36) does either, which maybe or may not be a good reference point.

Sorry for the 20 questions, I'm keen to see the content-type being set, just not sure which approach is correct/best.